### PR TITLE
fix: rest api errors

### PR DESCRIPTION
**Pull request by chaitanyaprem**
# Description
Fixes issues reported in #914 .
Also added additional logs for easy debugging.

# Changes

- [ ] Return 4xx instead of 5xx where appropriate
- [ ] Add some logs

# Tests

### CODE REVIEW

**Review by fbarbu15 — APPROVED** [review]

**Comment by chaitanyaprem** [comment]
@richard-ramos , not sure why but the postgres test keeps failing even if try to rerun the action.
Can you please take a look? I don't think it is realted to my changes.

File: cmd/waku/server/rest/relay.go [hunk 1]
```diff
@@ -113,13 +116,13 @@ func (r *RelayService) postV1Subscriptions(w http.ResponseWriter, req *http.Requ
 func (r *RelayService) getV1Messages(w http.ResponseWriter, req *http.Request) {
 	topic := topicFromPath(w, req, "topic", r.log)
 	if topic == "" {
-		return
+		topic = relay.DefaultWakuTopic
```
Comment (richard-ramos) [inline cmd/waku/server/rest/relay.go:119]:
```suggestion
		r.log.Debug("topic is not specified, using default waku topic")
		topic = relay.DefaultWakuTopic
```

File: cmd/waku/server/rest/relay.go [hunk 2]
```diff
@@ -160,35 +163,39 @@ func (r *RelayService) getV1Messages(w http.ResponseWriter, req *http.Request) {
 func (r *RelayService) postV1Message(w http.ResponseWriter, req *http.Request) {
 	topic := topicFromPath(w, req, "topic", r.log)
 	if topic == "" {
-		return
+		r.log.Debug("topic is not specified,using default waku topic")
```
Comment (richard-ramos) [inline cmd/waku/server/rest/relay.go:166]:
```suggestion
		r.log.Debug("topic is not specified, using default waku topic")
```

**Review by richard-ramos — APPROVED** [review]
LGTM

**Comment by richard-ramos** [comment]
> 2023/11/22 15:27:43 an error 'failed to connect to `host=localhost user=postgres database=template1`: failed to receive message (read tcp [::1]:58578->[::1]:5432: read: connection reset by peer)' while reseting the db

seems like we do not handle reconnects :( reopening https://github.com/waku-org/go-waku/issues/745

**Pull request merged**